newIDE/app/src/ObjectEditor/Editors/SpriteEditor/CollisionMasksEditor/CollisionMaskHelper.js

```diff
@@ -0,0 +1,76 @@
+// @flow
+import axios from 'axios';
+import optionalLazyRequire from '../../../../Utils/OptionalLazyRequire';
+import { checkIfCredentialsRequired } from '../../../../Utils/CrossOrigin';
+import Jimp from 'jimp/es';
+const lazyRequireFs = optionalLazyRequire('fs');
+
+const gd = global.gd;
+
+async function getJimpImage(pathToFile: string) {
+  try {
+    if (pathToFile.startsWith('http://') || pathToFile.startsWith('https://')) {
+      const response = await axios.get(pathToFile, {
+        responseType: 'arraybuffer',
+        withCredentials: checkIfCredentialsRequired(pathToFile),
+      });
+
+      const imageBuffer = response.data;
+      const jimpImage = await Jimp.read(imageBuffer);
+
+      return jimpImage;
+    } else {
+      // For some reason, calling directly Jimp.read with a path does not work
+      // in the renderer process of Electron. So we read the file ourselves
+      // and pass the ArrayBuffer to Jimp.
+      const fs = lazyRequireFs();
+      if (!fs) return null;
+      const imageBuffer = fs.readFileSync(pathToFile).buffer;
+      const jimpImage = await Jimp.read(imageBuffer);
+
+      return jimpImage;
+    }
+  } catch (error) {
+    console.error('Error fetching image:', error);
+    return null;
+  }
+}
+
+export const getMatchingCollisionMask = async (imageFile: string) => {
+  // First detect, the most left, right, top and bottom pixels that are not transparent.
+  // This will be used to crop the image.
+  if (!Jimp) return null;
+  const jimpImage = await getJimpImage(imageFile);
+  if (!jimpImage) return null;
+  const { width, height } = jimpImage.bitmap;
+  if (!width || !height) return null;
+  let minX = width;
+  let maxX = 0;
+  let minY = height;
+  let maxY = 0;
+  jimpImage.scan(0, 0, width, height, function(x, y, idx) {
+    const alpha = this.bitmap.data[idx + 3];
+    if (alpha > 0) {
```

Comment:
What if the sprite have some kind of embedded boom effect?
Should the threshold be at 25%?

Reply:
Good question, I haven't tested it on many images, just default ones.
Do you have a few "special" ones I can test with?

Reply:
I think it doesn't worth a manual test. It's more a choice to make.
I've seen users add bloom effect on their sprites with external tools because it's more optimized than using a shader.
This example have some in "Orb" and "ShotTrail" objects, but the alpha will depend on the style. So, I don't think it need to work on this particular example. 
https://gdevelop.io/game-example/game-feel-demo

I think 25% it about right as it won't:
- cut in many sprites
- not give too much margin on other.

Reply:
Let's keep it simple then to avoid 2 passes, I'll update the threshold to 64 (25% of 255)